NavViewsAndOptionSelector/KPIndexListTableViewController.m

```diff
@@ -0,0 +1,182 @@
+//
+//  KPIndexListTableViewController.m
+//  NavViewsAndOptionSelector
+//
+//  Created by MacMan on 8/14/15.
+//  Copyright (c) 2015 MacManApp. All rights reserved.
+//
+
+#import "KPIndexListTableViewController.h"
+#import "KPIndex.h"
+#import "KPAddItemTableViewController.h"
+#import "KPEditItemTableViewController.h"
+@interface KPIndexListTableViewController ()
+
+@end
+
+@implementation KPIndexListTableViewController
+
+@synthesize indexArray = _indexArray;
+
+
+
+
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    self.indexArray = [[NSMutableArray alloc] init];
+    
+    KPIndex *indexItem0 = [[KPIndex alloc] initWithName:@"use + to add an item" done:NO];
+    [self.indexArray addObject:indexItem0];
+    
+    
+    KPIndex *doneItem0 = [[KPIndex alloc] initWithName:@"click on item to edit it" done:YES];
+    [self.indexArray addObject:doneItem0];
+    
+    KPIndex *indexItem1 = [[KPIndex alloc] initWithName:@"swipe left on an item to delete" done:NO];
+    [self.indexArray addObject:indexItem1];
+    
+    KPIndex *indexItem2 = [[KPIndex alloc] initWithName:@"Item 2" done:NO];
+    [self.indexArray addObject:indexItem2];
+    
+    KPIndex *indexItem3 = [[KPIndex alloc] initWithName:@"Item 3" done:NO];
+    [self.indexArray addObject:indexItem3];
+    
+    KPIndex *indexItem4 = [[KPIndex alloc] initWithName:@"Item 4" done:NO];
+    [self.indexArray addObject:indexItem4];
+    
+    KPIndex *indexItem5 = [[KPIndex alloc] initWithName:@"Item 5" done:NO];
+    [self.indexArray addObject:indexItem5];
+    
+    KPIndex *indexItem6 = [[KPIndex alloc] initWithName:@"Item 6" done:NO];
+    [self.indexArray addObject:indexItem6];
+    
+    KPIndex *indexItem7 = [[KPIndex alloc] initWithName:@"Item 7" done:NO];
+    [self.indexArray addObject:indexItem7];
+    
+    KPIndex *indexItem8 = [[KPIndex alloc] initWithName:@"Item 8" done:NO];
+    [self.indexArray addObject:indexItem8];
+    
+    [self.tableView reloadData];
```

Comment:
You could make this code more concise by adding all of the elements at once instead of one-by-one, or even iterating through an array of strings, initializing your `KPIndex` object, and adding it to the array, like this:

```objective-c
self.indexArray = [[NSMutableArray alloc] init];

NSArray *indexNames = @[@"use + to add an item",
                        @"click on item to edit it",
                        @"swipe left on an item to delete",
                        @"Item 2",
                        @"Item 3",
                        @"Item 4",
                        @"Item 5",
                        @"Item 6",
                        @"Item 7",
                        @"Item 8"];

for (size_t i = 0; i < indexNames.count; i++) {
    KPIndex *item = [[KPIndex alloc] initWithName:indexNames[i] done:NO];
    [self.indexArray addObject:item];
}
```